test/type_test.ts

```diff
@@ -6,6 +6,11 @@ describe('types', () => {
      () => { expectTranslate('var x: foo.Bar;').to.equal(' foo . Bar x ;'); });
   it('drops type literals',
      () => { expectTranslate('var x: {x: string, y: number};').to.equal(' dynamic x ;'); });
+  it('translates string index signatures to dartisms', () => {
+    expectTranslate('var x: {[k: string]: any[]};')
```

Comment:
Well, that particular example isn't not allowed because all the properties
have to have the type that the indexer suggests. In fact, that's a useful
pattern, you can specify part of the object shape and allow "other" as well
using
`{size: number, [key: string]: number, height: number, width: number}'
and that matches {size: 3, height: 4, width: 5, depth: 6}

However, we would not be able to type such a thing in Dart, so you're right
that this PR is missing a constraint. I'll add that.

On Fri, Oct 2, 2015 at 6:08 PM Victor Berchet <notifications@github.com>
wrote:

> In test/type_test.ts
> <https://github.com/angular/ts2dart/pull/293#discussion_r41081808>:
>
> > @@ -6,6 +6,11 @@ describe('types', () => {
> >       () => { expectTranslate('var x: foo.Bar;').to.equal(' foo . Bar x ;'); });
> >    it('drops type literals',
> >       () => { expectTranslate('var x: {x: string, y: number};').to.equal(' dynamic x ;'); });
> > +  it('translates string index signatures to dartisms', () => {
> > +    expectTranslate('var x: {[k: string]: any[]};')
>
> @alexeagle <https://github.com/alexeagle> Would this PR allow 'var x:
> {size: number, [k: string]: any[]};' ?
>
> We should have test for that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/ts2dart/pull/293/files#r41081808>.
>